learningresources/tests/test_models.py

```diff
@@ -145,7 +145,7 @@ def test_static_asset_filename_length(self):
                 FILE_PATH_MAX_LENGTH -
                 # minus the length of the base path of the Django storage and
                 # the base path of the file location temporary dir
-                len(os.path.join(base_path, temp_dir_path)) -
+                len('/'.join([base_path, temp_dir_path])) -
```

Comment:
Here is the explanation.

The join function works as it should: the problem is that `temp_dir_path` starts with `/` so all the previous paths are ignored.
Honestly I thought that `os.path.join` was doing a simple string concatenation, so I must be careful in the future. 
```
def join(a, *p):
    """Join two or more pathname components, inserting '/' as needed.
    If any component is an absolute path, all previous path components
    will be discarded.  An empty last part will result in a path that
    ends with a separator."""
    path = a
    for b in p:
        if b.startswith('/'):
            path = b
        elif path == '' or path.endswith('/'):
            path +=  b
        else:
            path += '/' + b
    return path
```